src/ray/core_worker/core_worker.cc

```diff
@@ -3272,7 +3276,6 @@ void CoreWorker::HandleAssignObjectOwner(const rpc::AssignObjectOwnerRequest &re
 }
 
 void CoreWorker::YieldCurrentFiber(FiberEvent &event) {
-  RAY_CHECK(worker_context_.CurrentActorIsAsync());
```

Comment:
Could we keep `CurrentActorIsAsync`? Maybe we need to cache the result and reuse it elsewhere, like in `CoreWorker::Shutdown()`.